tests/unit/file_lib_test.c

```diff
@@ -1614,6 +1614,24 @@ static void test_file_read(void)
     return_to_test_dir();
 }
 
+static void test_read_file_stream_to_buffer(void)
+{
+    setup_tempfiles();
+
+    {
+        const size_t length = strlen(TEST_STRING);
+        char buf[1024] = {0};
+        FILE *const file = safe_fopen(TEST_FILE, "r");
+        assert_true(file != NULL);
+        const ssize_t read = ReadFileStreamToBuffer(file, length, buf);
+        fclose(file);
```

Comment:
Would be also nice to do a read again here to check that it fails correctly on a closed stream.

Reply:
I don't think this is defined behavior, or at least I couldn't find anything about calling fread on a closed file in the C standard, other than:

> The value of a pointer to a FILE object is indeterminate after the associated file is closed.

Maybe not a good idea to test this?

Reply:
OK